torch/_inductor/codegen/cpp.py

```diff
@@ -3962,10 +3962,19 @@ def get_indexing_ranges_exprs(node):
 
                 ref_node = node2 if len(vars1) < len(vars2) else node1
 
-                extra_indexing_constraints = get_indexing_ranges_exprs(ref_node)
+                ref_indexing_constraints = get_indexing_ranges_exprs(ref_node)
 
                 node_to_recomp.recompute_size_and_body(
-                    extra_indexing_constraints=extra_indexing_constraints
+                    extra_indexing_constraints=ref_indexing_constraints
+                )
+
+                node_to_recomp_indexing_constraints = get_indexing_ranges_exprs(
+                    node_to_recomp
+                )
+
+                # fix the case where ref_node also changes its ranges
+                ref_node.recompute_size_and_body(  # type: ignore[union-attr]
```

Comment:
Caling `recompute_size_and_body` may be expensive, if we can avoid to call it for cases when we do not need it...

Reply:
Thanks for the comment! Changed to call it only if `vars1 != vars2`.